blockchain/chain/src/store/chain_store.rs

```diff
@@ -300,23 +280,85 @@ where
     Ok(())
 }
 
-/// Gets 32 bytes of randomness for ChainRand paramaterized by the DomainSeparationTag, ChainEpoch, Entropy
-pub fn get_randomness<DB: BlockStore>(
+/// Finds the latest beacon entry given a tipset up to 20 blocks behind
+pub fn latest_beacon_entry<DB>(db: &DB, ts: &Tipset) -> Result<BeaconEntry, Error>
+where
+    DB: BlockStore,
+{
+    let check_for_beacon_entry = |ts: &Tipset| {
+        let cbe = ts.blocks()[0].beacon_entries();
+        if let Some(entry) = cbe.last() {
+            return Ok(Some(entry.clone()));
```

Comment:
You could call `cbe.pop()` here instead of `last()` to avoid a clone

Reply:
it's a slice, not a vec

Reply:
😬